### PR TITLE
T-15381 Add ability to set custom limits for enterprise plan

### PR DESCRIPTION
**Pull request by commoddity**
This PR is intended to support the ability to set custom daily relay limit for Enterprise plan type applications. It also cleans up some redundant duplicated data used on the Application struct's fields.

It adds an Enterprise plan type and updates the Application struct to use a Limits field that contains an AppLimits struct.

It also contains all necessary database changes to support the new `app_limits` table.

### CODE REVIEW

File: postgres-driver/application.go [hunk 1]
```diff
@@ -506,10 +586,14 @@ func (d *PostgresDriver) WriteApplication(app *repository.Application) (*reposit
 		return nil, ErrInvalidAppStatus
 	}
 
-	if !repository.ValidPayPlanTypes[app.PayPlanType] {
+	if !repository.ValidPayPlanTypes[app.Limit.PayPlan.Type] {
 		return nil, ErrInvalidPayPlanType
 	}
 
+	if app.Limit.PayPlan.Type != repository.Enterprise && app.Limit.CustomLimit != 0 {
```
Comment (adshmh) [inline postgres-driver/application.go]: Not a blocker, but this and other similar checks likely belong in the `Application` type itself

**Review by adshmh — CHANGES_REQUESTED** [review]
Could you please rebase the commits to remove code unrelated to PR.

File: repository/repository.go [hunk 2]
```diff
@@ -17,31 +18,72 @@ type Repository interface {
...
 
+func (a *Application) DailyLimit() int {
+	if a.Limit.PayPlan.Type == Enterprise {
+		return a.Limit.CustomLimit
+	}
+
+	return a.Limit.PayPlan.Limit
+}
+
+func (a *Application) IsInvalid() error {
```
Comment (adshmh) [inline repository/repository.go]: Not a blocker, but it is a bit non-conventional to use such a name. `Validate` may be a better choice.

File: repository/repository.go [hunk 3]
```diff
@@ -113,22 +162,36 @@ type AppLimits struct {
...
 	Status               AppStatus             `json:"status,omitempty"`
-	PayPlanType          PayPlanType           `json:"payPlanType,omitempty"`
 	FirstDateSurpassed   time.Time             `json:"firstDateSurpassed,omitempty"`
 	GatewaySettings      *GatewaySettings      `json:"gatewaySettings,omitempty"`
 	NotificationSettings *NotificationSettings `json:"notificationSettings,omitempty"`
+	Limit                *AppLimit             `json:"appLimit,omitempty"`
 	Remove               bool                  `json:"remove,omitempty"`
 }
 
+func (u *UpdateApplication) IsInvalid() error {
```
Comment (adshmh) [inline repository/repository.go]: Same comment as before regarding the naming, but still not a blocker.

File: repository/repository.go [hunk 4]
```diff
@@ -462,16 +525,17 @@ func loadData(file string, data interface{}) error {
 type Table string
 
 const (
+	TableLoadBalancers        Table = "loadbalancers"
```
Comment (adshmh) [inline repository/repository.go:528]: Another non-blocker comment, regarding this and all other diffs that show up due to simply moving where code/function is placed in the file: I recommend to keep these out of future feature-PRs, and to use separate (and much simpler and easier to read) PRs that just perform the reorganization of the code placement.

**Review by adshmh — DISMISSED** [review]
Just a few non-blocker comments to address later if we can. Otherwise LGTM. Thanks!

File: repository/repository.go (hunk 4 above)
Comment (commoddity) [reply repository/repository.go:528]: Agreed, I will do that in future PRs.

File: repository/repository.go (hunk 3 above)
Comment (commoddity) [reply repository/repository.go]: agreed, Validate is a better naming, I'll update before merging.

**Review by adshmh — APPROVED** [review]
Thank you for the update. LGTM

**Pull request merged**